### DISCUSSION TITLE
Feature Request: invalid workflow file should show as failed or errored in PR / commit checks

### DISCUSSION THREAD

**Discussion by jszwedko** [community/community#25739] [Actions]
<p>I recently introduced an error in a workflow file in a PR but neglected to notice as the PR showed as green and passing all checks (it just hadn’t run the jobs from that workflow).</p>
<p>It would be nice to surface invalid workflow files as a failing PR check so that it it is less likely that they will be missed. Some other CI systems report this an on “error” rather than a “failure” to distinguish them.</p>
<p>Thanks!</p>

**Comment by ghost** [comment, marked as answer]
<p>Thank you for your feedback - we appreciate it and will pass it on to the Product team.</p>
<p>For future reference – we really appreciate feedback on how we can make GitHub even better and the best way to report any feature requests is directly to us through our feedback form: <a href="https://support.github.com/contact/feedback">https://support.github.com/contact/feedback</a></p>
<p>Our roadmap is not publicly visible, so we recommend that you keep an eye on the <a href="https://github.com/blog">GitHub Blog</a> for the latest announcements about new features.</p>

**Comment by bandlerr** [comment]
<p>It’s been about a year since this was brought up. Is this still on the radar? An invalid workflow file <em>should</em> show up as a failure in PR/push checks. Show the green “all checks passes” when your workflow itself has failed is clearly misleading and erroneous.</p>

**Comment by boulderZ** [comment]
<p>I have same question? I can not find a fix for this. It makes catching problems very difficult and could allow a merge to happen that should not happen. I would think that this should have been a high priority to fix.</p>

**Comment by naikrovek** [comment]
<p>maybe the <code>act</code> tool can help.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/nektos/act">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/nektos/act" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/284498/181092962-6d6e55cd-6bea-4c2a-b871-00abe7f066d2.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/1/510009a7e5dba890d6fe8a78b80c5bdc809f6429_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/1/510009a7e5dba890d6fe8a78b80c5bdc809f6429_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/1/510009a7e5dba890d6fe8a78b80c5bdc809f6429.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/1/510009a7e5dba890d6fe8a78b80c5bdc809f6429_2_10x10.png"></div>

<h3><a href="https://github.com/nektos/act" target="_blank" rel="noopener nofollow ugc">GitHub - nektos/act: Run your GitHub Actions locally 🚀</a></h3>

  <p>Run your GitHub Actions locally 🚀. Contribute to nektos/act development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>you may be able to run that in “dry-run” mode as part of a PR checker and then flag the PR as valid or not according to the output.</p>
<p>I don’t know the specifics of any of this but I wanna help, and this seems like it can help.</p>
